components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -925,30 +931,141 @@ public UserRecoveryData getUserRecoveryData(String code, RecoverySteps step) thr
+                            recoveryFlowId);
+        }
+        return recoveryData;
+    }
+
+    /**
+     * Get user recovery flow data using the recovery flow id.
+     *
+     * @param recoveryDataDO User Recovery Data object.
+     * @throws IdentityRecoveryException If an error occurred while validating the recoveryId.
+     */
+    public UserRecoveryFlowData loadUserRecoveryFlowData(UserRecoveryData recoveryDataDO)
+            throws IdentityRecoveryException {
+
+        UserRecoveryFlowData userRecoveryFlowData;
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        try {
+            userRecoveryFlowData = userRecoveryDataStore.loadRecoveryFlowData(recoveryDataDO);
+        } catch (IdentityRecoveryException e) {
+            // Map code expired error to new error codes for user account recovery.
+            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_FLOW_ID.getCode().equals(e.getErrorCode())) {
+                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_FLOW_ID.getCode());
+            } else if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_FLOW_ID.getCode().equals(e.getErrorCode())) {
+                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_RECOVERY_FLOW_ID.getCode());
+            } else {
+                e.setErrorCode(Utils.prependOperationScenarioToErrorCode(e.getErrorCode(),
+                        IdentityRecoveryConstants.USER_ACCOUNT_RECOVERY));
+            }
+            throw e;
+        }
+        if (userRecoveryFlowData == null) {
+            throw Utils.handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_RECOVERY_FLOW_DATA,
+                    recoveryDataDO.getRecoveryFlowId());
+        }
+        return userRecoveryFlowData;
+    }
+
+    /**
+     * Update recovery OTP attempt.
+     *
+     * @param recoveryFlowId Recovery Flow Id.
+     * @param failedAttempts Failed Attempts.
+     */
+    public void updateRecoveryDataFailedAttempts(String recoveryFlowId, int failedAttempts) throws IdentityRecoveryException {
+
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        userRecoveryDataStore.updateFailedAttempts(recoveryFlowId, failedAttempts);
+    }
+
+    /**
+     * Update recovery OTP resend count.
+     *
+     * @param recoveryFlowId Recovery Flow Id.
+     * @param resendCount    Current Resend Count.
+     */
+    public void updateRecoveryDataResendCount(String recoveryFlowId, int resendCount) throws IdentityRecoveryException {
+
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        userRecoveryDataStore.updateCodeResendCount(recoveryFlowId, resendCount);
+    }
+
+    /**
+     * Invalidate the recovery Data.
+     *
+     * @param recoveryFlowId Recovery Flow Id.
+     */
```

Comment:
thrown exception is not mentioned in the method comment.
Check all java doc comments of newly introduced methods

Reply:
Fixed with 4c8c9f64b52364e581f4f4f1890afe0f11577c08